manifests/init.pp

```diff
@@ -541,6 +541,16 @@
 #
 #
 class kubernetes (
+  Enum['archive','package'] $kubernetes_install_method  = 'package',
+  Optional[String] $kubernetes_archive                  = "kube-bins-${kubernetes_version}.tar.gz",
+  Optional[String] $kubernetes_archive_checksum                  = undef,
+  Optional[String] $kubernetes_source                   = undef,
+  Optional[String] $crictl_archive                  = "crictl-v1.22.0-linux-amd64.tar.gz",
+  Optional[String] $crictl_archive_checksum         = undef,
+  Optional[String] $crictl_source                   = undef,
+  Optional[String] $cni_plugins_archive                  = "cni-plugins-linux-amd64-v0.8.2.tgz",
```

Comment:
If you assign a value other than undef to the variable, the `Optional[]` datatype doesn't make any sense, you can replace it with `String[]`

Reply:
examples have been moved to comment, should the appear in the docstrings above inside init.pp?